### PR TITLE
Better message when no GPU is found.

### PR DESCRIPTION
**Pull request by trivialfis**
https://github.com/dmlc/xgboost/issues/5592

### CODE REVIEW

**Comment by trivialfis** [comment]
@RAMitchell In a longer term,  I would like to have something similar to: https://www.tensorflow.org/api_docs/python/tf/config/list_physical_devices in XGBoost, WDYT?

**Comment by RAMitchell** [comment]
@trivialfis seems like users could do this with many available third-party tools, why does xgboost need to do this?

**Comment by trivialfis** [comment]
@RAMitchell We can use it in combined with many other things:

- Is XGBoost compiled with CUDA?
- What version of CUDA is XGBoost compiled with?
- Is XGBoost compiled with OpenMP?
- What's the default number of threads in XGBoost: #4975
- Verbosity #5011 .
- Is CPU prefetching enabled in XGBoost?

**Comment by trivialfis** [comment]
@hcho3 @RAMitchell Could you please review?

File: src/learner.cc [hunk 1]
```diff
@@ -187,8 +187,12 @@ void GenericParameter::ConfigureGpuId(bool require_gpu) {
   // raw model), number of available GPUs could be different.  Wrap around it.
   int32_t n_gpus = common::AllVisibleGPUs();
   if (n_gpus == 0) {
+    if (gpu_id != kCpuId) {
+      LOG(WARNING) << "No visible GPU is found, setting `gpu_id` to -1";
+    }
     this->UpdateAllowUnknown(Args{{"gpu_id", std::to_string(kCpuId)}});
   } else if (gpu_id != kCpuId && gpu_id >= n_gpus) {
+    LOG(WARNING) << "Only " << n_gpus << " GPUs are visible, setting `gpu_id` to " << gpu_id;
```
Comment (hcho3) [inline src/learner.cc:195]:
```suggestion
    LOG(WARNING) << "Only " << n_gpus << " GPUs are visible, setting `gpu_id` to " << (gpu_id % n_gpus);
```

File: tests/python-gpu/load_pickle.py [hunk 2]
```diff
@@ -49,3 +55,16 @@ def test_wrap_gpu_id(self):
         test_x = xgb.DMatrix(x)
         res = bst.predict(test_x)
         assert len(res) == 10
+
+    def test_training_on_cpu_only_env(self):
+        assert os.environ['CUDA_VISIBLE_DEVICES'] == '-1'
+        print('RUnning')
```
Comment (hcho3) [inline tests/python-gpu/load_pickle.py:61]:
```suggestion
```

**Pull request merged**